client/checkout/utils/upe.js

```diff
@@ -279,12 +305,16 @@ export const getPaymentMethodTypes = ( paymentMethodType ) => {
 };
 
 /**
- * Returns the value of the email input on the blocks checkout page.
+ * Returns the email value from store API.
  *
- * @return {string} The value of email input.
+ * @return {string} The email value.
  */
 export const getBlocksEmailValue = () => {
-	return document.getElementById( 'email' ).value;
+	// .wcpay-payment-element container is rendered only when new payment method is selected
+	return document
+		.querySelector( '.wcpay-payment-element' )
+		?.closest( 'form' )
+		?.querySelector( '#billing_email' )?.value;
```

Comment:
In the block-based checkout, I see that the email input has `id` attribute of `email`. I don't see a `billing_email` input.
<img width="1020" alt="Screenshot 2024-12-02 at 10 29 49 AM" src="https://github.com/user-attachments/assets/b8536b27-c039-4c66-b166-42749d2fafc4">

Reply:
I confused this with the shortcode checkout elements, sorry about that. Moved to the correct ID in 70d2f84f735d7a9615a480081ea3e68378354a82